tests/eval_files/async102.py

```diff
@@ -265,3 +266,20 @@ async def foo_nested_async_for():
                 j
             ) in trio.bypasslinters:
                 ...
+
+
+# nested funcdef (previously false alarm)
+async def foo_nested_funcdef():
+    try:
+        ...
+    except:
+
+        async def foobar():
+            await foo()
+
+
+async def foo_nested_funcdef():
+    try:
+        ...
+    except:
+        x = lambda: await foo()
```

Comment:
`SyntaxError: 'await' outside async function` - there's no such thing as an async lambda.

We might want to check that our test files are syntactically valid to avoid any subtle issues in future?

Reply:
After testing, it appears that [`ast.parse`](https://docs.python.org/3/library/ast.html#ast.parse) does not pick up on it - and neither mypy nor ruff has any checks for it. Calling [`compile`](https://docs.python.org/3/library/functions.html#compile) on it does surface the error, as it does additional scoping checks that `ast.parse` does not do.

Reply:
(more obvious errors have been caught in the past by mypy/ruff). Opened #268